### PR TITLE
Bugfix/json messages

### LINKED ISSUE #250: Proofread and correct descriptions of ENA model
#### Description

It would be nice to see the tooltip help texts in UI more descriptive instead of repeating the title or label of the field.  Here it would be useful to know the meaning of the fields/description, we could use a technical writer to proofread the descriptions.

In short replace non helpful description text with more informative descriptions.

These description texts come from ENA model and are at json files at https://github.com/CSCfi/metadata-submitter/tree/master/metadata_backend/helpers/schemas : 

- ena_analysis.json
- ena_dac.json
- ena_dataset.json
- ena_experiment.json
- ena_policy.json
- ena_run.json
- ena_sample.json
- ena_study.json

#### DoD (Definition of Done)

User can see more helpful tooltips at the UI, when ENA model schema's descriptions are proofread.

#### Testing

A  review of UI.

### PR DESCRIPTION
**Pull request by blankdots**
### Related issues

Closes #250

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Changes Made

1. add JSON schema spelling checker to pyspelling github action
2. optimise wordlist by adding regex ignore patterns
3. fix current spelling mistakes
4. added pyspelling to pre-commit hooks (fixed syntax for scripts according to https://github.com/koalaman/shellcheck )
5. enum are sorted alphabetically, with the exception of `other` and `unspecified` values which are left at the end of the list
6. add descriptions to JSON schemas
7. allow for `accession` key in `referenceAlignment` & process `sequence` as array, previously all `accession` keys were converted to `accessionId` which is not correct
8. add default `gender` as `unknown`

### Testing

- [x] Tests do not apply

### Mention

This might cause breaking changes to front-end tests

### CODE REVIEW

File: .spellcheck.yml [hunk 1]
```diff
@@ -8,6 +8,40 @@ matrix:
...
+      - open: '<([A-Za-z0-9-_:.]+)|(https?://[^\\s/$.?#].[^\\s]+|[A-Za-z0-9-_:.]+)'
+        close: '>'
   sources:
   - 'docs/*.rst'
+  - '**/*.md'
+  default_encoding: utf-8
+
+- name: JSON schemas
+  aspell:
+    lang: en
```
Comment (genie9) [inline .spellcheck.yml:28]: I think we could add `ignore-case: true`, which would shorten our wordlist.

Comment (blankdots) [reply .spellcheck.yml:28]: ok will add that as well

**Review by genie9 — APPROVED** [review]

**Review by genie9 — APPROVED** [review]

**Comment by blankdots** [comment]
changes in front-end should be introduced in https://github.com/CSCfi/metadata-submitter-frontend/issues/622

**Review by teemukataja — APPROVED** [review]

File: metadata_backend/helpers/schemas/ena_policy.json [hunk 2]
```diff
@@ -282,29 +290,30 @@
...
         "dacRef": {
             "title": "Data Access Committee Reference",
-            "description": "Identifies the data access committee to which this policy pertains.",
+            "description": "Identifies the data access committee to which this Policy pertains.",
             "$ref": "#/definitions/reference"
         },
         "policy": {
             "title": "Policy",
-            "description": "Identifies the data access committee to which this policy pertains.",
+            "description": "Policies can be added either by adding the text of the Policy of by pointing to an existing URL.",
```
Comment (genie9) [inline metadata_backend/helpers/schemas/ena_policy.json:311]: ...text of the Policy `or` by pointing to an ...

**Review by genie9 — APPROVED** [review]
Added a few comments on spelling and approved PR. GJ!

File: metadata_backend/helpers/schemas/datacite.json [hunk 3]
```diff
@@ -629,56 +646,59 @@
...
+                            "Is Variant Form Of",
+                            "Is Original Form Of",
+                            "Is Identical To"
                         ]
                     },
                     "relatedMetadataScheme": {
                         "type": "string",
-                        "title": "Related Metadata Scheme"
+                        "title": "Related Metadata Scheme",
+                        "description": "To be Use only with this relation pair: (Has Metadata/Is Metadata For). Example DDI-L"
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schemas/datacite.json]: good catch, fixed.

File: metadata_backend/helpers/schemas/folders.json [hunk 4]
```diff
@@ -255,55 +262,61 @@
               "affiliation": {
                 "type": "array",
                 "title": "Affiliations",
+                "description": "The organizational or institutional affiliation of the creator. Start writing the name of the organization or institution and a suggestion will be made from Research Organization Registry (ROR) Community API.",
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schemas/folders.json]: added.

File: metadata_backend/helpers/schemas/folders.json [hunk 5]
```diff
@@ -641,7 +657,7 @@
             "Vietnamese",
             "Volapük",
             "Walloon",
-            "Wolof",
+            "WolOf",
```
[the remark this answers is not in the document]
Comment (blankdots) [reply metadata_backend/helpers/schemas/folders.json]: seems like a typo, fixed.

File: metadata_backend/helpers/schemas/ena_policy.json (hunk 2 above)
Comment (blankdots) [reply metadata_backend/helpers/schemas/ena_policy.json]: good catch, fixed.

**Comment by hannyle** [comment]
Could this be rebased on `develop` branch so it can include the fix in PR https://github.com/CSCfi/metadata-submitter/pull/333 ?

**Pull request merged**